### PR TITLE
Updates to timer categories

### PR DESCRIPTION
**Pull request by sebastiangrimberg**
Added a few new categories for timing, including specific to linear solver vs. eigensolver vs. time stepping, and other related changes. This improves the organization and use of the timer classes throughout the code.

### CODE REVIEW

**Comment by sebastiangrimberg** [comment]
@kllrak, I just requested review from you because you had done the initial work on the timer. Perhaps the more relevant PR for your review is https://github.com/awslabs/palace/pull/189, where the changes here are documented.

**Comment by kllrak** [comment]
Looks like changes are adjustments to timing categories and where the timers are used, rather than the timing mechanism. In that case I'm probably not the most efficient reviewer here, but thanks for the heads up.

File: palace/drivers/drivensolver.cpp [hunk 1]
```diff
@@ -365,11 +355,17 @@ ErrorIndicator DrivenSolver::SweepAdaptive(SpaceOperator &spaceop, PostOperator
...
     omega += delta_omega;
   }
+  BlockTimer bt0(Timer::POSTPRO);
+  SaveMetadata(prom.GetLinearSolver());
   return indicator;
 }
 
 int DrivenSolver::GetNumSteps(double start, double end, double delta) const
 {
+  if (end < start)
```
Comment (hughcars) [inline palace/drivers/drivensolver.cpp:365]: Maybe `<=`? Skips the logic if driving a single frequency.

Comment (sebastiangrimberg) [reply palace/drivers/drivensolver.cpp:365]: My thought was that the logic below works fine for `start = end`, so no need to bypass it.

**Review by hughcars — APPROVED** [review]

**Pull request merged**